pkg/util/httputil/client.go

```diff
@@ -0,0 +1,52 @@
+package httputil
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"net/http"
+	"net/url"
+
+	"os"
+
+	"github.com/authgear/authgear-server/pkg/lib/config"
+)
+
+func NewHTTPClient(env *config.EnvironmentConfig) *http.Client {
+	client := http.DefaultClient
```

Comment:
Do not modify DefaultClient. According to https://pkg.go.dev/net/http#Client a zero value of Client is good enough.